server/tests/steps/test_waterfall.py

```diff
@@ -153,3 +153,58 @@ def test_bug_duplicate_rows():
     )
 
     assert_dataframes_equals(result_df, expected_df)
+
+
+def test_waterfall_bug_drill():
```

Comment:
Mayeb it would be nice to add a commentary explaining this bug:
```suggestion
def test_waterfall_bug_drill():
```
```suggestion
def test_waterfall_bug_drill():
"""
Tuple (label, parent) should be unique only among one "group by" sub-df.
"""
```